File: src/main/java/no/nav/domain/pensjon/regler/grunnlag/PersonDetalj.java
package no.nav.domain.pensjon.regler.grunnlag;

import no.nav.domain.pensjon.regler.PenPerson;
import no.nav.domain.pensjon.regler.kode.BorMedTypeCti;
import no.nav.domain.pensjon.regler.kode.GrunnlagKildeCti;
import no.nav.domain.pensjon.regler.kode.GrunnlagsrolleCti;
import no.nav.domain.pensjon.regler.kode.SivilstandTypeCti;

import java.io.Serializable;
import java.util.Date;

/**
 * PersonDetalj inneholder persondetaljer som er relevante for fastsettelse av vedtak i Pensjonsløsningen,
 * dvs detaljer om den faktiske rollen en bruker har i et krav, sivilstand , pensjonsfaglig vurdert
 * sivilstand og barn for en definert periode.
 */
public class PersonDetalj implements Serializable {
    /**
     * Rollen denne personen har i kontekst av kravet.
     */
    private GrunnlagsrolleCti grunnlagsrolle;
    /**
     * Fra-og-med dato for rollens gyldighet.
     */
    private Date rolleFomDato;
    /**
     * Til-og-med dato for rollens gyldighet.
     */
    private Date rolleTomDato;

    /**
     * Representerer personens sivilstand i henhold til TPS.
     */
    private SivilstandTypeCti sivilstandType;
    /**
     * Eventuell angivelse av hvilken annen person som sivilstandType relaterer seg til,
     * for eksempel ektefelle eller samboer.
     */
    private PenPerson sivilstandRelatertPerson;
    /**
     * Representerer om og hvordan personen bor sammen med en annen person med persongrunnlag på kravet.
     * Refereres konseptuelt som pensjonsfaglig vurdert sivilstand.
     */
    private BorMedTypeCti borMed;
    /**
     * Detaljer om barnet hvis rolle=BARN. Angir om barnet bor med en annen forelder.
     */
    private BarnDetalj barnDetalj;

    /**
     * Angir om det er opprettet barne- eller ektefelletillegg kravlinje for persongrunnlaget
     * med gitt rolle.
     */
    private boolean tillegg;
    /**
     * Angir om persondetaljen brukes som grunnlag på kravet.
     */
    private boolean bruk;
    /**
     * Angir kilden til persondetaljen.
     */
    private GrunnlagKildeCti grunnlagKilde;

    private Boolean serskiltSatsUtenET;

    private Boolean epsAvkallEgenPensjon;

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/grunnlag/PersonDetalj.kt
package no.nav.pensjon.regler.domain.grunnlag

import no.nav.pensjon.regler.domain.PenPerson
import no.nav.pensjon.regler.domain.enum.BorMedTypeEnum
import no.nav.pensjon.regler.domain.enum.GrunnlagkildeEnum
import no.nav.pensjon.regler.domain.enum.GrunnlagsrolleEnum
import no.nav.pensjon.regler.domain.enum.SivilstandEnum
import java.io.Serializable
import java.util.*

/**
 * PersonDetalj inneholder persondetaljer som er relevante for fastsettelse av vedtak i Pensjonsløsningen,
 * dvs detaljer om den faktiske rollen en bruker har i et krav, sivilstand , pensjonsfaglig vurdert
 * sivilstand og barn for en definert periode.
 */
open class PersonDetalj : Serializable {
    /**
     * Rollen denne personen har i kontekst av kravet.
     */
    var grunnlagsrolleEnum: GrunnlagsrolleEnum? = null

    /**
     * Fra-og-med dato for rollens gyldighet.
     */
    var rolleFomDato: Date? = null

    /**
     * Til-og-med dato for rollens gyldighet.
     */
    var rolleTomDato: Date? = null

    /**
     * Representerer personens sivilstand i henhold til TPS.
     */
    var sivilstandTypeEnum: SivilstandEnum? = null

    /**
     * Eventuell angivelse av hvilken annen person som sivilstandType relaterer seg til,
     * for eksempel ektefelle eller samboer.
     */
    var sivilstandRelatertPerson: PenPerson? = null

    /**
     * Representerer om og hvordan personen bor sammen med en annen person med persongrunnlag på kravet.
     * Refereres konseptuelt som pensjonsfaglig vurdert sivilstand.
     */
    var borMedEnum: BorMedTypeEnum? = null

    /**
     * Detaljer om barnet hvis rolle=BARN. Angir om barnet bor med en annen forelder.
     */
    var barnDetalj: BarnDetalj? = null

    /**
     * Angir om det er opprettet barne- eller ektefelletillegg kravlinje for persongrunnlaget
     * med gitt rolle.
     */
    var tillegg = false

    /**
     * Angir om persondetaljen brukes som grunnlag på kravet.
     */
    var bruk: Boolean = true

    /**
     * Angir kilden til persondetaljen.
     */
    var grunnlagKildeEnum: GrunnlagkildeEnum? = null

    var serskiltSatsUtenET: Boolean? = null
    var epsAvkallEgenPensjon: Boolean? = null
}